packages/quick_actions/quick_actions_ios/pubspec.yaml

```diff
@@ -13,7 +13,7 @@ flutter:
     implements: quick_actions
     platforms:
       ios:
-        pluginClass: FLTQuickActionsPlugin
+        pluginClass: QuickActionsPlugin
```

Comment:
Don't we still require the thin ObjC wrapper around Swift plugins? That's still what we are generating in the template; I thought it was required for the Swift-plugin-in-ObjC host to work in iOS apps. @jmagman

Reply:
The generated file is this: 

```
#import "GeneratedPluginRegistrant.h"

#if __has_include(<integration_test/IntegrationTestPlugin.h>)
#import <integration_test/IntegrationTestPlugin.h>
#else
@import integration_test;
#endif

#if __has_include(<quick_actions_ios/QuickActionsPlugin.h>)
#import <quick_actions_ios/QuickActionsPlugin.h>
#else
@import quick_actions_ios;
#endif

@implementation GeneratedPluginRegistrant

+ (void)registerWithRegistry:(NSObject<FlutterPluginRegistry>*)registry {
  [IntegrationTestPlugin registerWithRegistrar:[registry registrarForPlugin:@"IntegrationTestPlugin"]];
  [QuickActionsPlugin registerWithRegistrar:[registry registrarForPlugin:@"QuickActionsPlugin"]];
}

@end
```

It is directly using `[QuickActionsPlugin registerWithRegistrar:]`. So we should be fine here?

Reply:
@jmagman Did `__has_include` resolve all the issues with direct Swift-into-Obj-C import? If so, we should fix the plugin template to stop generating the Obj-C wrapper.

Reply:
Are you suggesting to remove `GeneratedPluginRegistrant.h/m` and move the logic inside the `AppDelegate`?

Reply:
No, I'm talking about the ObjC files [here](https://github.com/flutter/flutter/tree/master/packages/flutter_tools/templates/plugin/ios-swift.tmpl/Classes).

It seems like either there is still at least one combination of app language and library build type that requires those wrappers, in which case this plugin needs them too, or that's no longer the case, in which case the template shouldn't need them.

Reply:
Gotcha. I did not find any clue about why we needed this wrapper from [the original PR](https://github.com/flutter/flutter/pull/10259)

The wrapped swift class is an NSObject subclass which is already exposed to objc runtime.  

- objc plugin + objc/swift app: unchanged
- swift plugin + objc app: `GeneratedPluginRegistrant.m` directly uses `QuickActionsPlugin.swift` as shown in this example
- swift plugin + swift app: shouldn't involve any objc logic at all. 

So I don't think the ObjC wrapper is necessary.

Reply:
@jmagman Did extensive investigation a while back about various combinations of host app language, plugin language, and compilation mode (static library vs dynamic library), so should know if there are still edge cases that require it or not.

Reply:
Objc-plugin-in-Swift-app will work without the wrapper if the Objective-C plugins define modules correctly, which they should out of the box as of plugin podspec template change in https://github.com/flutter/flutter/pull/40302 (Flutter 1.10.3).  Plugins created before that point may not work when used in a Swift app since it's missing `DEFINES_MODULE`.  I think all other combinations should work without the wrapper.

We could add a warning to the tool for the plugin author that they need to add `{ 'DEFINES_MODULE' => 'YES' }` to their podspec (although I don't think we have a good tool hook to know they are building a plugin example app and not just a regular app).  Or we could add some documentation about how to fix up the podspec when their plugin customers report it doesn't work in a Swift app.

Reply:
@jmagman thanks for the explanation. Sounds like we should remove it. Created an issue here https://github.com/flutter/flutter/issues/111928

Reply:
> Are all the languages swapped in this comment by accident, or we talking about different cases? I was talking about the ObjC wrapper that we generate in the Swift plugin template, to make them look like older (ObjC) plugins, which IIRC was necessary for the Swift-plugin-in-ObjC-app case (for at least one of the library build modes).

You're right I was mixing this up with the Objective-C files that get generated in a Swift iOS app, which I think was originally so the Swift app can import Objective-C plugins pre-modules.
https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app_shared/ios-swift.tmpl/Runner/Runner-Bridging-Header.h
https://github.com/flutter/flutter/blob/5aad12cab1e9c0863ea5fbc9a770b2734d7772db/packages/flutter_tools/lib/src/flutter_plugins.dart#L456

<img width="242" alt="Screen Shot 2022-09-19 at 3 58 35 PM" src="https://user-images.githubusercontent.com/682784/191133717-c4331d07-4e58-4a56-bbe0-bc727cf4dec0.png">

But you're right, I _think_ the Swift plugins also don't need the Objective-C code now.  Fortunately this area is pretty well tested, so if someone tries https://github.com/flutter/flutter/issues/111928 the tests should let us know if there are any corner cases I'm not remembering.

Reply:
Reading it again (@jmagman you gave me hints about clang module), i think it's possible that the original plugin template wanted to expose both header and module, so that its objc client can freely choose from importing headers vs importing modules. This benefit seems so trivial that we should probably remove the template.